### DISCUSSION TITLE
Populate Tree Items <FluentTreeView>

### DISCUSSION THREAD

**Discussion by chrisonmoon** [microsoft/fluentui-blazor#2048] [Q&A] [answered]
Hello,

I have a question about populating a FluentTreeview. So far I have the following code, which populates the root and the children. This works fine so far. 
<img width="368" alt="image" src="https://github.com/microsoft/fluentui-blazor/assets/60311234/f7be9434-ccff-44fa-b59d-4ec0a10ce6d3">

<img width="598" alt="image" src="https://github.com/microsoft/fluentui-blazor/assets/60311234/20f9c951-1d5b-4797-97ec-417fa9e12a4f">

<img width="554" alt="image" src="https://github.com/microsoft/fluentui-blazor/assets/60311234/53c32a4d-92f4-4231-ac2b-68f996028339">

<img width="689" alt="image" src="https://github.com/microsoft/fluentui-blazor/assets/60311234/df9dd126-64d4-48b2-bdad-42f835efd1d3">

What i would like to have is the following:
<img width="606" alt="image" src="https://github.com/microsoft/fluentui-blazor/assets/60311234/630a29fb-3015-4298-b130-1f4adf9da232">

I tried to add FluentCheckbox to my rendertree but it did not work: 

builder.OpenComponent<FluentTreeItem>(0);
builder.AddAttribute(1, nameof(FluentTreeItem.Text), NodeID.FullName);
builder.AddAttribute(2, nameof(FluentTreeItem.ChildContent), BuildTreeItemFragment(NodeID));
builder.OpenComponent<FluentCheckbox>(3);
..................             
builder.CloseComponent();
builder.CloseComponent();

What am I doing wrong ? 

Thanks a lot Chris

**Comment by dvoituron** [comment]
Probably that will be easier in some days, using this new `ItemTemplate` feature :-) See #1945

**Reply by chrisonmoon** [reply]
Hi - i've seen that.  would be very nice.

Thanks :-)

**Reply by chrisonmoon** [reply]
Hello Denis - you have any guess, when the feature is available ?

Thanks a lot Chris

**Comment by chrisonmoon** [comment]
Hello Denis,

you have a guess, when the feature will be available? 

Thank you , Chris

**Reply by dvoituron** [reply, marked as answer]
You can already used it using our Preview feed: https://github.com/microsoft/fluentui-blazor/blob/dev/docs/using-latest-daily.md

A new final version will be published soon

**Reply by chrisonmoon** [reply]
Thank you :-)
